package/gluon-mesh-vpn-wireguard/luasrc/lib/gluon/upgrade/399-mesh-vpn-wireguard-from-fastd

```diff
@@ -0,0 +1,16 @@
+#!/usr/bin/lua
+
+local uci = require('simple-uci').cursor()
+local util = require 'gluon.util'
+
+local wg_private_key = uci:get("network_gluon-old", 'wg_mesh', "private_key")
+local fastd_secret = uci:get('fastd', 'mesh_vpn', 'secret')
+
+if (not wg_private_key or not wg_private_key:match("^" .. ("[%a%d+/]"):rep(42) .. "[AEIMQUYcgkosw480]=$"))
+    and fastd_secret and fastd_secret:match(('%x'):rep(64)) then
```

Comment:
```
if (wg_private_key and wg_private_key_match) then return end
  return
end

if not fastd or not fastd_secret:match(('%x'):rep(64)) then
    return
end

....
```

Reply:
resolved in 531bbbb41a5c6b20e80526cd9a30cfe41ee4f88b